Controller/ResourceController.php

```diff
@@ -32,21 +34,34 @@ public function __construct(
 
     public function resourceAction($repositoryName, $path)
     {
-        $repository = $this->registry->get($repositoryName);
-        $resource = $repository->get('/' . $path);
-
-        $context = SerializationContext::create();
-        $context->enableMaxDepthChecks();
-        $context->setSerializeNull(true);
-        $json = $this->serializer->serialize(
-            $resource,
-            'json',
-            $context
-        );
-
-        $response = new Response($json);
-        $response->headers->set('Content-Type', 'application/json');
-        
+        try {
+            $repository = $this->registry->get($repositoryName);
+            $resource = $repository->get('/' . $path);
+
+            $context = SerializationContext::create();
+            $context->enableMaxDepthChecks();
+            $context->setSerializeNull(true);
+            $json = $this->serializer->serialize(
+                $resource,
+                'json',
+                $context
+            );
+
+            $response = new Response($json);
+            $response->headers->set('Content-Type', 'application/json');
+        } catch (\Exception $e) {
+            $code = 500;
+
+            if ($e instanceof ResourceNotFoundException) {
+                $code = 404;
+            }
+
+            $response = new JsonResponse(array(
+                'exception' => get_class($e),
+                'message' => $e->getMessage()
+            ), $code);
+        }
+
         return $response;
```

Comment:
@WouterJ what about this question by dan? does it not happen automatically already? is it maybe only a symfony 2.3 issue?